src/realm/object-store/impl/collection_change_builder.cpp

```diff
@@ -147,6 +149,8 @@ void CollectionChangeBuilder::merge(CollectionChangeBuilder&& c)
         collection_root_was_deleted = true;
     }
 
+    collection_was_cleared = c.collection_was_cleared;
```

Comment:
This results is inconsistent behavior. If a clear is followed by an insert in a single write transaction it'll report `collection_was_cleared=true` plus an insertion, but if they're in two separate transactions that get coalesced it'll report `collection_was_cleared=false` plus an insertion. Whether or not this should stay set if there's subsequent modifications comes down to what the SDK needs, but it does need to behave the same way in both cases.

Reply:
I think that in this case what counts is the final state of the collection, so in this case probably having `collection_was_cleared=false` is the right choice.

Reply:
Agree - I think we should only report `collection_was_cleared` if the final state of the collection is empty. So an insertion should reset the `collection_was_cleared` flag to `false`. Might be a good idea to add an explicit test for that.